pylint/checkers/typecheck.py

```diff
@@ -806,6 +813,10 @@ def _is_invalid_isinstance_type(arg: nodes.NodeNG) -> bool:
         return False
     if isinstance(inferred, astroid.Instance) and inferred.qname() == BUILTIN_TUPLE:
         return False
+    if PY310_PLUS and isinstance(inferred, bases.UnionType):
```

Comment:
```suggestion
    if isinstance(inferred, bases.UnionType):
```
Is the `bases.UnionType` the legacy Union/Optional type? if so it is supported from 3.7+ I think.

Reply:
No, `Union[int, str]` is an instance of `ClassDef` and not `UnionType`. I'm not sure what exactly `UnionType` corresponds to.